routing/transit_world_graph.hpp

```diff
@@ -47,8 +48,7 @@ class TransitWorldGraph final : public WorldGraph
   RouteWeight CalcSegmentWeight(Segment const & segment) override;
   RouteWeight CalcLeapWeight(m2::PointD const & from, m2::PointD const & to) const override;
   bool LeapIsAllowed(NumMwmId mwmId) const override;
-
-  static bool IsTransitSegment(Segment const & segment);
+  std::unique_ptr<TransitInfo> GetTransitInfo(Segment const & segment) override;
```

Comment:
Can we use your wrapper here?

Reply:
Yes, but I'm not sure it's a good plan. We have simple interface for RouteSegment:
```C++
void SetTransitInfo(std::unique_ptr<TransitInfo> transitInfo);
bool HasTransitInfo() const;
TransitInfo const & GetTransitInfo() const;
```
It works like you expect it to work and you don't need to keep in mind the fact that we have some wrapper inside to make it copyable and small.
Using this wrapper outside of RouteSegment does not give any significant profit and makes us keep one more class in mind.